### ISSUE TITLE
Null attribute causes errors

### ISSUE THREAD

**Issue by pinguxx** [WebReflection/hyperHTML#32]
Im getting an error if i pass a null attribute

Uncaught TypeError: Cannot read property 'nodeType' of null
    at populateNode (test-bundle.js:643)
    at Array.any (test-bundle.js:506)

in a template like this
```
<div>
    <span>${data.placeholder}</span>
    more text
</div>
```

where placeholder can be null, i can do this

```
<div>
    <span>${data.placeholder || ''}</span>
    more text
</div>
```

i wonder if you can update hyperhtml where if the value its null or undefined the textcontent will be ''

maybe add an extra check for nulls/undefined in this function
https://github.com/WebReflection/hyperHTML/blob/master/hyperhtml.js#L130

Thanks

**Comment by WebReflection** [comment]
Are you trying to set text or HTML? If text, put at least one space around.

About null, I don't know if I want to end up handling dirty data. Why would have a null for a layout that accepts text on nodes or arrays as explained by the API? Those are also not the only place that need special handling for dirty data.

I'll think about it but it might be a won't-fix unless you provide me a good reason for accepting null values

**Comment by joshgillies** [comment]
Agree with you @WebReflection.

@pinguxx If you really want to solve the above issue might I suggest something I've been experimenting with inside `hypercomponents` is effectively wrapping `hyperHTML.wire()` in a function like:

```js
function wrappedWire (...args) {
  return hyperHTML.wire(this).apply(hyperHTML, args.map((arg) =>
    typeof arg === 'undefined' || arg === null ? '' : arg))
}

// use it just like you would with any old wire
wrappedWire`<div>${ maybeUndefined }</div>`
```

**Comment by WebReflection** [comment]
that's cool, but it also covers only one scenario.

`hyperHTML` templates accept 2 kind of values and their Array counterpart:

  1. a string, a DOM node (including document fragments)
  2. an array of strings, an Array of DOM nodes

Accepting dirty data means that I need to loop over all lists to see if one of those values and I don't think the job of this library is to sanitize its own signature.

So far I've built many examples, including a whole [TodoMVC](https://github.com/WebReflection/hypermvc/tree/master) and the whole [Hacker News](https://github.com/WebReflection/viper-news) and dirty data was unnecessary.

If you **really** need to handle that case, you can always explicitly opt in via `${model.prop || ''}` whenever it's appropriate, which should be usually not the default behavior.

TL;DR unless there's evidence that everyone needs `hyperHTML` to be fatter and slower due dirty data, this can be considered a won't fix.
